src/utils/git.js

```diff
@@ -0,0 +1,42 @@
+import _ from 'lodash'
+import cp from 'child_process'
+import Promise from 'bluebird'
+
+export const execFile = Promise.promisify(cp.execFile)
+
+export function parseDiffRanges(diff) {
+  const matches = diff.match(/^@@ -\d+,\d+ \+(\d+),(\d+) @@/gm)
+  if (!_.isEmpty(matches)) {
+    return matches.map(match => {
+      const [start, end] = /^@@ -\d+,\d+ \+(\d+),(\d+) @@/.exec(match).slice(1, 3)
+      return [parseInt(start, 10), parseInt(start, 10) + parseInt(end, 10)]
+    })
+  }
+  return []
+}
+
+const filenameRegex = /a\/([^\n]+) b\/[^\n]+/
+export function parseDiffForFile(diff) {
+  const matches = filenameRegex.exec(diff)
+  if (matches === null) {
+    return null
+  }
+  const filename = matches[1]
+  return { filename, ranges: parseDiffRanges(diff) }
+}
+
+export function parseFullDiff(diff) {
+  return _(diff.split('diff --git '))
```

Comment:
Ah, right. Didn't think of that when writing the comment on `filenameRegex` above (it doesn't matter if it skips the first one unless you also add `^` to that).

Then I suggest using a regex, which seems a bit nicer than adding a newline:
```
diff.split(/^diff --git /m)
```